src/clog/ob_log_file_pool.cpp

```diff
@@ -417,19 +419,54 @@ int ObLogWriteFilePool::create_tmp_file(const file_id_t file_id, char* fname, co
 int ObLogWriteFilePool::update_free_quota(const char* path, const int64_t percent, const int64_t limit_percent)
 {
   int ret = OB_SUCCESS;
-  struct statfs fsst;
+  int64_t used_size = 0;
+
   if (NULL == path || 0 > percent || 100 < percent || 0 > limit_percent || 100 < limit_percent) {
     ret = OB_INVALID_ARGUMENT;
-  } else if (OB_UNLIKELY(0 != statfs(path, &fsst))) {
+  } else if (OB_FAIL(get_total_used_size(used_size))) {
+    ret = OB_IO_ERROR;  
+    COMMON_LOG(ERROR, "get_total_used_size fail", K(ret));
+  } else if(OB_FAIL(calculate_free_quota(path, used_size, percent, limit_percent))){
     ret = OB_IO_ERROR;
-    CLOG_LOG(ERROR, "statfs error", K(ret), K(path), K(errno), KERRMSG);
-  } else {
-    const int64_t total_size = (int64_t)fsst.f_bsize * (int64_t)fsst.f_blocks;
-    const int64_t free_quota =
-        (int64_t)fsst.f_bsize * ((int64_t)fsst.f_blocks * percent / 100LL - (int64_t)(fsst.f_blocks - fsst.f_bavail));
-    const int64_t used_size = (int64_t)fsst.f_bsize * (int64_t)(fsst.f_blocks - fsst.f_bavail);
-    int64_t limit_quota = (int64_t)fsst.f_bsize *
-                          ((int64_t)fsst.f_blocks * limit_percent / 100LL - (int64_t)(fsst.f_blocks - fsst.f_bavail));
+    CLOG_LOG(ERROR, "calculate free quota error", K(ret), K(path), K(errno), KERRMSG);
+  } 
+  return ret;
+}
+
+int ObLogWriteFilePool::init_free_quota(const char* path, const int64_t percent, const int64_t limit_percent)
+{
+  int ret = OB_SUCCESS;
+  const int64_t used_size = 0;
+
+  if (NULL == path || 0 > percent || 100 < percent || 0 > limit_percent || 100 < limit_percent) {
+    ret = OB_INVALID_ARGUMENT;
+  } else if(OB_FAIL(calculate_free_quota(path, used_size, percent, limit_percent))){
+    ret = OB_IO_ERROR;
+    CLOG_LOG(ERROR, "calculate free quota error", K(ret), K(path), K(errno), KERRMSG);
+  }
+  return ret;
+}
+
+int ObLogWriteFilePool::calculate_free_quota(const char* path, const int64_t used_size, const int64_t percent, const int64_t limit_percent)
+{
+  int ret = OB_SUCCESS;
+  struct statvfs svfs;
+  const int64_t clog_disk_limit_size = ObServerConfig::get_instance().clog_disk_limit_size;
+  int64_t total_size = 0;
+
+  if (OB_UNLIKELY(0 != statvfs(path, &svfs))) {
+    ret = OB_IO_ERROR;
+    CLOG_LOG(ERROR, "statvfs error", K(ret), K(path), K(errno), KERRMSG);
+  } else{
+    if (clog_disk_limit_size != 0){
+      total_size = clog_disk_limit_size;
+    }else{
+      total_size = (int64_t)svfs.f_bsize * (int64_t)svfs.f_blocks;
+    }
+    
+    const int64_t free_quota = total_size * percent / 100LL - used_size;
```

Comment:
seem that used_size is not init

Reply:
used_size is a parameter of the function "calculate_free_quota", it is inited by the caller(update_free_quota or init_free_quota)

Reply:
I see that the use_size you pass in when initializing the clog is 0, and the use_size you pass in when updating is the size of the clog files, but note that the use_size of the disk where the clog is located during init is not necessarily 0, it may be used by other resources, so here i suggested you need to manually calculate the use_size of the disk. Similarly, the use_size passed in when update_free_quota cannot simply get the file size of the clog. You should also calculate the use_size of the actual disk (clog disk).